src/components/text-editor/text-editor.component.tsx

```diff
@@ -1,537 +1,279 @@
+  onLinkAdded?: (link: string, state: string) => void;
+  /** The callback to fire when the Save button within the editor is pressed */
+  onSave?: (value: SaveCallbackProps) => void;
+  /** The placeholder to display when the editor is empty */
+  placeholder?: string;
+  /** An array of link preview nodes to render in the editor */
+  previews?: React.JSX.Element[];
+  /** Whether the editor is read-only */
+  readOnly?: boolean;
+  /** Whether the content of the editor is required to have a value */
   required?: boolean;
-  /** Flag to configure component as optional. */
-  isOptional?: boolean;
-  /** Message to be displayed when there is an error */
-  error?: string;
-  /** Message to be displayed when there is a warning */
-  warning?: string;
-  /** [Legacy] Message to be displayed when there is an info */
-  info?: string;
-  /** Number greater than 2 multiplied by line-height (21px) to override the default min-height of the editor */
+  /** Number greater than 2 multiplied to override the default min-height of the editor */
   rows?: number;
-  /** The previews to display of any links added to the Editor */
-  previews?: React.ReactNode;
-  /** Callback to report a url when a link is added */
-  onLinkAdded?: (url: string) => void;
-  /** Hint text to be rendered when validationRedesignOptIn flag is set */
-  inputHint?: string;
+  /** The message to be shown when the editor is in an warning state */
+  warning?: string;
+  /** The initial value of the editor, as a HTML string, or JSON */
+  value?: string | undefined;
 }
 
-export const TextEditor = React.forwardRef<Editor, TextEditorProps>(
-  (
-    {
-      characterLimit = 3000,
-      labelText,
-      onChange,
-      value,
-      required,
-      error,
-      warning,
-      info,
-      toolbarElements,
-      rows,
-      previews,
-      onLinkAdded,
-      inputHint,
-      isOptional,
-      ...rest
-    }: TextEditorProps,
-    ref,
-  ) => {
-    const { validationRedesignOptIn } = useContext(NewValidationContext);
-    const [isFocused, setIsFocused] = useState(false);
-    const [inlines, setInlines] = useState<InlineStyleType[]>([]);
-    const [activeInlines, setActiveInlines] = useState<
-      Partial<Record<InlineStyleType, boolean>>
-    >({});
-    const [focusToolbar, setFocusToolbar] = useState(false);
-
-    const editorRef = useRef<Editor>(null);
-    const wrapper = useRef<HTMLDivElement>(null);
-    const editor = ref || editorRef;
-    const contentLength = getContent(value).getPlainText("").length;
-    const moveCursor = useRef(contentLength > 0);
-    const lastKeyPressed = useRef<null | string>();
-    const inputHintId = useRef(`${guid()}-hint`);
-    const { current: id } = useRef(guid());
-
-    const { labelId, validationId, ariaDescribedBy } = useInputAccessibility({
-      id,
-      validationRedesignOptIn,
-      error,
-      warning,
-      info,
-      label: labelText,
-    });
+export const CreateFromHTML = (html: string) => {
```

Comment:
comment (non-blocking): it might be worth having these in separate file/s as they're not used in the component etc so can be tree-shaken if they aren't used